contracts/src/main/kotlin/com/contractsAndStates/contracts/LongestRoadContract.kt

```diff
@@ -0,0 +1,57 @@
+package com.contractsAndStates.contracts
+
+import com.contractsAndStates.states.*
+import net.corda.core.contracts.CommandData
+import net.corda.core.contracts.Contract
+import net.corda.core.contracts.requireSingleCommand
+import net.corda.core.contracts.requireThat
+import net.corda.core.transactions.LedgerTransaction
+import java.security.PublicKey
+
+class LongestRoadContract : Contract {
+
+    companion object {
+        const val ID = "com.contractsAndStates.contracts.LongestRoadContract"
+    }
+
+    override fun verify(tx: LedgerTransaction) {
+        val command = tx.commands.requireSingleCommand<Commands>()
+
+        val longestRoadInputStates = tx.inputsOfType<LongestRoadState>().first()
+        val roadsInputStates = tx.inputsOfType<RoadState>()
+        val settlementsInputStates = tx.inputsOfType<SettlementState>()
+        val inputGameBoardState = tx.inputsOfType<GameBoardState>().single()
+
+        val outputLongestRoadState = tx.outputsOfType<LongestRoadState>().first()
+
+        when (command.value) {
+            is Commands.Claim -> requireThat {
+
+                /**
+                 *  ******** BUSINESS LOGIC ********
+                 */
+
+                val candidate = longestRoad(
```

Comment:
After merge, this function will probably end being one in `GameBoardState`.

Reply:
I don't like much the idea of "God" state. I will make it very hard to manage in the future.

Reply:
Look at how you can delegate the interface to an element of the class:
https://github.com/rogersanick/SettlersOfCordan/pull/31/files#diff-e4a04c60ddcc70a5eaf8e66144dd8683R33
That simplifies a lot.

Reply:
What exactly should I look at? 
It is a very large PR

Reply:
Look at these lines 70-76 https://github.com/rogersanick/SettlersOfCordan/blob/fdc83bc5392190d669a6cb5ceb6051f440e20f48/contracts/src/main/kotlin/com/contractsAndStates/states/GameBoardState.kt#L70-L76